dcc-submission-server/src/main/java/org/icgc/dcc/submission/server/repository/SubmissionRepository.java

```diff
@@ -0,0 +1,116 @@
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.icgc.dcc.submission.server.repository;
+
+import static org.icgc.dcc.submission.release.model.QSubmission.submission;
+
+import java.util.Collection;
+import java.util.List;
+
+import lombok.NonNull;
+import lombok.val;
+
+import org.icgc.dcc.submission.release.model.QSubmission;
+import org.icgc.dcc.submission.release.model.Submission;
+import org.icgc.dcc.submission.release.model.SubmissionState;
+import org.mongodb.morphia.Datastore;
+import org.mongodb.morphia.Morphia;
+import org.mongodb.morphia.query.Query;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.mysema.query.mongodb.MongodbQuery;
+
+public class SubmissionRepository extends AbstractRepository<Submission, QSubmission> {
+
+  @Autowired
+  public SubmissionRepository(Morphia morphia, Datastore datastore) {
+    super(morphia, datastore, submission);
+  }
+
+  public void addSubmissions(@NonNull Iterable<Submission> submissions) {
+    save(submissions);
+  }
+
+  public void addSubmission(@NonNull Submission submission) {
+    save(submission);
+  }
+
+  public void updateExistingSubmissions(@NonNull Iterable<Submission> submissions) {
+    for (val submission : submissions) {
+      updateFirst(
+          createFilterByIdQuery(submission),
+          submission,
+          false);
+    }
+  }
+
+  public int updateSubmission(@NonNull Submission submission) {
+    val result = updateFirst(createFilterByIdQuery(submission), submission, false);
+
+    return result.getUpdatedCount();
+  }
+
+  public List<Submission> findSubmissionStateByReleaseName(@NonNull String releaseName) {
+    return where(entity.releaseName.eq(releaseName))
+        .list(entity.state);
+  }
+
+  public Submission findSubmissionByReleaseNameAndProjectKey(@NonNull String releaseName, @NonNull String projectKey) {
+    return createFilterByReleaseNameQuery(releaseName)
+        .where(entity.projectKey.eq(projectKey)).singleResult();
+  }
+
+  public List<Submission> findSubmissionsByReleaseNameAndProjectKey(@NonNull String releaseName,
+      @NonNull Collection<String> projectKeys) {
+    return createFilterByReleaseNameQuery(releaseName)
+        .where(entity.projectKey.in(projectKeys))
+        .list();
+  }
+
+  public List<Submission> findSubmissionsByReleaseName(@NonNull String releaseName) {
+    return createFilterByReleaseNameQuery(releaseName).list();
+  }
+
+  public List<Submission> findSubmissions() {
+    return list();
+  }
+
+  public List<Submission> findSubmissionsByProjectKey(@NonNull String projectKey) {
+    return query()
+        .where(entity.projectKey.eq(projectKey))
+        .list();
+  }
+
+  public int deleteByReleaseAndNotState(@NonNull String releaseName, @NonNull SubmissionState state) {
+    val result = delete(createQuery()
+        .filter("releaseName == ", releaseName)
```

Comment:
I think 'no'. I didn't find how Morphia DSL could be used for DML. It looks like the version we use supports querying only.